### ISSUE TITLE
Changing cstate for better latency

### ISSUE THREAD

**Issue by tanvp112** [bottlerocket-os/bottlerocket#3079]
**What I'd like:**

I would like to achieve what is documented [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/processor_state_control.html#c-states), but I noticed **settings.boot.kernel-parameters** are ignored for non-metal instance type such as c6i.16xlarge and found:

_Please note that boot settings currently only exist for the bare metal variants and *-k8s-1.23 variants. Boot settings will be added to any future variant introduced after Bottlerocket v1.8.0._

Our use case doesn't go up to metal type. Is there a plan to implement this feature soon?

**Any alternatives you've considered:**

EKS optimized AMI.

**Comment by bcressey** [comment]
The documentation does not make this totally clear, but the variants where these settings work are:
- `metal-k8s-*`
- `aws-k8s-*` except `aws-k8s-1.22`
- `vmware-k8s-*` except `vmware-k8s-1.22`

"Metal" here is a platform designation and refers to properties such as how the host is provisioned with first boot data, how the networking is configured, etc.

All EC2 instance types including `*.metal` use one of the `aws-*` variants.

In short as long as your clusters are running at least Kubernetes 1.23 then the feature should work.

**Comment by tanvp112** [comment]
Let's take compute instance type as example. I have tested on EKS 1.26 with **c6i.16xlarge** (managed node group), everything works as expected until I checked in sheltie (/proc/cmdline) and found that `processor.max_cstate` was not added to the boot parameters. You meant to say this is because the minimum instance type requirement is **c6i.metal**? If true, can we have support for non-metal instance type like **c6i.16xlarge**?

**Comment by zmrow** [comment]
@tanvp112 Can you share the userdata you supplied to the instance?  It should work on the variant you mentioned.

[`settings.boot.reboot-to-reconcile`](https://github.com/bottlerocket-os/bottlerocket#boot-related-settings) is the key here.  The instance needs to reboot to take advantage of the boot config settings, since they are applied via userdata after the instance has already booted the first time.
